### PR TITLE
Make SonarCloud URL customizable

### PR DESCRIPTION
**Pull request by damien-urruty-sonarsource**

### CODE REVIEW

File: common/src/main/java/org/sonarlint/intellij/common/util/SonarLintUtils.java [hunk 1]
```diff
@@ -47,8 +48,9 @@
 public class SonarLintUtils {
 
   private static final Logger LOG = Logger.getInstance(SonarLintUtils.class);
-  private static final String[] SONARCLOUD_ALIAS = {"https://sonarqube.com", "https://www.sonarqube.com",
-    "https://www.sonarcloud.io", "https://sonarcloud.io"};
+  public static final String SONARCLOUD_URL = System.getProperty("sonarlint.internal.sonarcloud.url", "https://sonarcloud.io");
+  private static final Set<String> SONARCLOUD_ALIAS = Set.copyOf(List.of("https://sonarqube.com", "https://www.sonarqube.com",
```
Comment (nquinquenel) [inline common/src/main/java/org/sonarlint/intellij/common/util/SonarLintUtils.java:52]: Why not directly create a Set with `Set.of`?

File: build.gradle.kts [hunk 2]
```diff
@@ -171,6 +171,7 @@ val runIdeDirectory: String by project
 
 tasks.runIde {
     systemProperty("sonarlint.telemetry.disabled", "true")
+    //systemProperty("sonarlint.internal.sonarcloud.url", "https://sonarcloud.io/")
```
Comment (nquinquenel) [inline build.gradle.kts:174]: Is there a reason to keep this comment?

**Review by nquinquenel — APPROVED** [review]
LGTM, just few questions

File: common/src/main/java/org/sonarlint/intellij/common/util/SonarLintUtils.java (hunk 1 above)
Comment (damien-urruty-sonarsource) [reply common/src/main/java/org/sonarlint/intellij/common/util/SonarLintUtils.java:52]: `Set.of` throws when there are duplicates in the arguments. What I want to do here is to keep `https://sonarcloud.io` in all cases, plus the custom one that might be configured

File: build.gradle.kts (hunk 2 above)
Comment (damien-urruty-sonarsource) [reply build.gradle.kts:174]: this is how you activate the custom URL. I will add a comment to explain that

**Review by nquinquenel — APPROVED** [review]

**Pull request merged**